pkg/apis/core/v1/storagecluster.go

```diff
@@ -360,6 +360,8 @@ type CloudStorageCapacitySpec struct {
 
 // CloudStorageSpec details of storage in cloud environment for entire cluster
 type CloudStorageSpec struct {
+	// Provider spec for the cloud provider, such as GKE, AWS etc
+	Provider *string `json:"cloudProvider,omitempty"`
```

Comment:
We need to rename the json field name to `provider`.
Also change the crd schema file above.

Reply:
Thanks for catching it.